OpenRA.Mods.Common/ServerTraits/LobbyCommands.cs

```diff
@@ -124,9 +124,10 @@ static void CheckAutoStart(S server)
 				if (server.LobbyInfo.Slots.Any(sl => sl.Value.Required && server.LobbyInfo.ClientInSlot(sl.Key) == null))
 					return;
 
-				// Can't have insufficient spawns
-				var availableSpawnPointCount = server.Map.SpawnPoints.Length - server.LobbyInfo.DisabledSpawnPoints.Count;
-				if (availableSpawnPointCount < server.LobbyInfo.Clients.Count(c => !c.IsObserver))
+				// Make sure we don't have insufficient spawns when a map needs to use them (defines more than 0)
+				var spawnPoints = server.Map.SpawnPoints.Length;
+				var availableSpawnPoints = spawnPoints - server.LobbyInfo.DisabledSpawnPoints.Count;
```

Comment:
Strictly speaking, this should really be `server.Map.SpawnPoints.Except(server.LobbyInfo.DisabledSpawnPoints).Count()` if we want to be robust against invalid values in the `DisabledSpawnPoints` list.

Reply:
I get a CS1929 because `CPos[]` does not expose `Except`.

Reply:
Ah right, `server.Map.SpawnPoints` is a `CPos[]` while `server.LobbyInfo.DisabledSpawnPoints` is a `List<int>`...

This would need to be `Enumerable.Range(1, spawnPoints + 1).Except(server.LobbyInfo.DisabledSpawnPoints).Count()` or `spawnPoints - server.LobbyInfo.DisabledSpawnPoints.Where(x => x > 0 && x <= spawnPoints).Distinct().Count()` to have a chance of working.